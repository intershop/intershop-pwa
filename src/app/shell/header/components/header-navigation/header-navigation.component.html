<ul class="navbar-nav main-navigation-list d-md-none" [ngClass]="{ accordion: view !== 'full' }">
  <ng-container *ngTemplateOutlet="categoryTemplate"></ng-container>
</ul>
<ul class="navbar-nav main-navigation-list d-none d-md-block" [ngClass]="{ accordion: view === 'small' }">
  <ng-container *ngTemplateOutlet="categoryTemplate"></ng-container>
</ul>
<!--- Category Template -->
<ng-template #categoryTemplate>
  <li
    *ngFor="let category of categories"
    #submenu
    class="dropdown"
    [ngClass]="{ open: isOpened(category) }"
    (mouseover)="subMenuShow(submenu)"
    (mouseout)="subMenuHide(submenu)"
    (click)="subMenuHide(submenu)"
  >
    <a
      [routerLink]="['/category', category.uniqueId]"
      [attr.data-testing-id]="category.uniqueId + '-link'"
      [ngStyle]="{ width: !category.hasChildren() ? '100%' : '' }"
    >
      {{ category.name }}
    </a>
    <a
      *ngIf="category.hasChildren()"
      class="dropdown-toggle"
      [ngClass]="{ 'd-none': view === 'full', 'd-md-none': view === 'auto' }"
      (click)="toggleOpen(category)"
    >
      <fa-icon *ngIf="isOpened(category); else closed" [icon]="['fas', 'minus']"></fa-icon>
      <ng-template #closed> <fa-icon [icon]="['fas', 'plus']"></fa-icon> </ng-template>
    </a>
    <ish-sub-category-navigation
      [category]="category"
      [subCategoriesDepth]="1"
      [view]="view"
    ></ish-sub-category-navigation>
  </li>
</ng-template>
