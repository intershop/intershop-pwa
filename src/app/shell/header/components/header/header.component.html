<div class="header container">
  <ul *ngIf="!isSticky" class="user-links">
    <li *ngIf="!isSticky && deviceType !== 'mobile'">
      <ish-login-status-container data-testing-id="user-status-desktop"></ish-login-status-container>
    </li>
    <li *ishFeature="'compare'" class="d-none d-md-block">
      <ish-product-compare-status-container></ish-product-compare-status-container>
    </li>
  </ul>

  <div class="header-utility">
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container *ngIf="!isSticky"></ish-language-switch-container>
    </div>
    <ish-login-status-container *ngIf="isSticky || deviceType === 'mobile'" view="small"></ish-login-status-container>

    <ish-mini-basket-container
      view="full"
      class="d-none"
      [ngClass]="isSticky ? 'd-lg-block' : 'd-md-block'"
    ></ish-mini-basket-container>
    <ish-mini-basket-container
      view="small"
      [ngClass]="isSticky ? 'd-lg-none' : 'd-md-none'"
    ></ish-mini-basket-container>
  </div>

  <div *ngIf="!isSticky" class="search-container header-search-container d-none d-md-block">
    <ish-search-box-container
      [configuration]="{
        placeholderText: 'search.searchbox.instructional_text' | translate,
        autoSuggest: true,
        maxAutoSuggests: 10
      }"
    ></ish-search-box-container>
  </div>

  <div class="logo-wrapper">
    <a
      *ngIf="(!isSticky && deviceType === 'tablet') || deviceType === 'pc'"
      rel="home"
      routerLink="/home"
      class="logo"
      data-testing-id="header-home-link-desktop"
    >
      <img src="assets/img/logo.png" alt="Logo" />
    </a>
    <a
      *ngIf="(isSticky && deviceType !== 'pc') || deviceType === 'mobile'"
      rel="home"
      routerLink="/home"
      class="mobile-logo"
      data-testing-id="header-home-link-mobile"
      ><img src="assets/img/logo_mobile.png" alt="Logo" width="41" height="41"
    /></a>
  </div>

  <button class="navbar-toggler" type="button" (click)="navbarCollapsed = !navbarCollapsed">
    <span class="sr-only">{{ 'common.button.navbarCollapsed.text' | translate }}</span>
    <div class="hamburger-menu" [ngClass]="{ open: showNavBar }">
      <!--- strokes of the hamburger menu -->
      <span></span> <span></span> <span></span>
    </div>
  </button>

  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md" [ngbCollapse]="navbarCollapsed">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container>
      </div>
    </nav>
  </div>
</div>
