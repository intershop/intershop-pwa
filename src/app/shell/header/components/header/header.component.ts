import { isPlatformBrowser } from '@angular/common';
import {
  ChangeDetectionStrategy,
  Component,
  HostListener,
  Inject,
  Input,
  OnDestroy,
  OnInit,
  PLATFORM_ID,
} from '@angular/core';
import { NavigationStart, Router } from '@angular/router';
import { Subject } from 'rxjs';
import { filter, takeUntil } from 'rxjs/operators';

import { MEDIUM_BREAKPOINT_WIDTH } from 'ish-core/configurations/injection-keys';

/**
 * The Header Component displays the page header.
 *
 * It uses the {@link LoginStatusContainerComponent} for rendering the users login status.
 * It uses the {@link ProductCompareStatusContainerComponent} for rendering the product compare button and count.
 * It uses the {@link MobileBasketContainerComponent} for rendering the mobile basket button and basket item count.
 * It uses the {@link LanguageSwitchContainerComponent} for rendering the language selection dropdown.
 * It uses the {@link SearchBoxContainerComponent} for rendering the search box.
 * It uses the {@link HeaderNavigationContainerComponent} for rendering the pages main navigation.
 * It uses the {@link MiniBasketContainerComponent} for rendering mini basket on desktop sized viewports.
 *
 * @example
 * <ish-header></ish-header>
 */
@Component({
  selector: 'ish-header',
  templateUrl: './header.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class HeaderComponent implements OnInit, OnDestroy {
  @Input() isSticky = false;

  navbarCollapsed = false;
  showSearch = false;
  screenHeight: number;
  mobileStickyHeaderHeight = 40;

  private destroy$ = new Subject();

  constructor(
    @Inject(MEDIUM_BREAKPOINT_WIDTH) private mediumBreakpointWidth: number,
    @Inject(PLATFORM_ID) private platformId: string,
    private router: Router
  ) {}

  ngOnInit() {
    if (isPlatformBrowser(this.platformId)) {
      this.navbarCollapsed = window.innerWidth < this.mediumBreakpointWidth;
    }

    // collapse mobile menu on router navigation start event
    // TODO: check testing and router subscription vs. ngrx state handling
    this.router.events
      .pipe(
        filter(event => event instanceof NavigationStart),
        takeUntil(this.destroy$)
      )
      .subscribe(() => {
        if (isPlatformBrowser(this.platformId) && window.innerWidth < this.mediumBreakpointWidth) {
          this.navbarCollapsed = true;
        }
      });
  }

  ngOnDestroy() {
    this.destroy$.next();
  }

  @HostListener('window:resize', ['$event'])
  mobileViewHandler(event) {
    this.navbarCollapsed = event.target.innerWidth < this.mediumBreakpointWidth;
    this.screenHeight = event.target.innerHeight - this.mobileStickyHeaderHeight;
  }

  toggleSearch(): void {
    this.showSearch = !this.showSearch;
  }
}
