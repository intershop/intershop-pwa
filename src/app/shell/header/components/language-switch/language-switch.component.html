<div *ngIf="locale" class="language-switch" [ngClass]="view" ngbDropdown placement="bottom-right">
  <a ngbDropdownToggle class="language-switch-link">
    <fa-icon [icon]="['fas', 'globe-americas']" class="header-icon"></fa-icon>
    <span class="language-switch-current-selection d-inline">{{ view==="accordion"?locale.displayName: locale.value }}<span class="switch_arrow"></span></span>
  </a>
  <div ngbDropdownMenu class="language-switch-container dropdown-menu" style="left:0px!important">
    <div class="language-switch-menu-container">
      <ul *ngIf="availableLocales && availableLocales.length">
        <ng-container *ngFor="let l of availableLocales">
          <li *ngIf="l.lang !== locale.lang" >
          <a (click)="switch(l)">
            {{ l.displayName }}
          </a>
        </li>
        </ng-container>
      </ul>
    </div>
  </div>
</div>
