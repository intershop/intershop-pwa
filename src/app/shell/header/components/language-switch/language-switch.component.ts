import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output } from '@angular/core';

import { Locale } from 'ish-core/models/locale/locale.model';

@Component({
  selector: 'ish-language-switch',
  templateUrl: './language-switch.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class LanguageSwitchComponent {
  @Input() locale: Locale;
  @Input() availableLocales: Locale[];
  @Input() view: '' | 'accordion' = '';
  @Output() localeChange = new EventEmitter<Locale>();

  switch(locale: Locale) {
    this.localeChange.emit(locale);
    return false; // prevent actual navigation, only change localized values on the current page and set language state
  }
}
