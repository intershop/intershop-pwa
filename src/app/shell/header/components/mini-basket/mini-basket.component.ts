import { Location } from '@angular/common';
import { ChangeDetectionStrategy, Component, Input, OnChanges, SimpleChanges } from '@angular/core';

import { BasketView } from 'ish-core/models/basket/basket.model';
import { HttpError } from 'ish-core/models/http-error/http-error.model';

/**
 * The Mini Basket Component displays a quick overview over the users basket items.
 * It uses the {@link ProductImageComponent} for the rendering of product images.
 *
 * @example
 * <ish-mini-basket
      [basket]="basket$ | async"
      [view]="view"
      [basketAnimation]="basketAnimation$ | async"
      [error]="basketError$ | async"
    ></ish-mini-basket>
 */
@Component({
  selector: 'ish-mini-basket',
  templateUrl: './mini-basket.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class MiniBasketComponent implements OnChanges {
  /**
   * The basket that should be displayed.
   */
  @Input() basket: BasketView;
  @Input() view: 'auto' | 'small' | 'full' = 'auto';
  @Input() basketAnimation = '';
  @Input() error: HttpError;

  isCollapsed = true;
  itemCount = 0;

  constructor(private location: Location) {}

  ngOnChanges(c: SimpleChanges) {
    this.animateBasket(c);
    if (this.basket) {
      this.itemCount = this.basket.totalProductQuantity;
    } else {
      this.itemCount = 0;
    }
  }

  /**
   * Animate basket after the product line item count has changed
   * Open basket if an error occured
   */
  animateBasket(c: SimpleChanges) {
    if (c && c.basketAnimation && this.location.path() !== '/basket') {
      if (this.basketAnimation && c.basketAnimation.currentValue === 'tada') {
        this.open();
      } else {
        this.collapse();
      }
    }
    if (c && !!c.error && this.error) {
      this.open();
    }
  }

  /**
   * Toggle the collapse state of the mini basket programmatically.
   */
  toggleCollapse() {
    this.isCollapsed = !this.isCollapsed;
  }

  /**
   * Collapse the mini basket programmatically.
   */
  collapse() {
    this.isCollapsed = true;
  }

  /**
   * Open the mini basket programmatically.
   */
  open() {
    this.isCollapsed = false;
  }
}
