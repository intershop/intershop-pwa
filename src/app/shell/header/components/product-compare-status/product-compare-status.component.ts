import { ChangeDetectionStrategy, Component, Input } from '@angular/core';

@Component({
  selector: 'ish-product-compare-status',
  templateUrl: './product-compare-status.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class ProductCompareStatusComponent {
  @Input() productCompareCount: number;

  @Input() view: 'auto' | 'small' | 'full' = 'auto';
}
