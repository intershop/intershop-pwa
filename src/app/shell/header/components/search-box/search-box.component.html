<!-- NEEDS_WORK: review and adapt -->
<form
  role="search"
  class="search clearfix"
  name="SearchBox_Header"
  [formGroup]="searchForm"
  (ngSubmit)="submitSearch()"
>
  <input
    [placeholder]="configuration?.placeholderText || ''"
    formControlName="search"
    autocomplete="off"
    name="SearchTerm"
    type="text"
    class="form-control searchTerm"
    (focus)="hidePopup()"
    (blur)="hidePopup()"
    (keydown.esc)="hidePopup()"
    (keydown.arrowleft)="selectSuggestedTerm(-1)"
    (keydown.arrowright)="selectSuggestedTerm(-1)"
    (keydown.arrowdown)="selectSuggestedTerm(activeIndex + 1)"
    (keydown.arrowup)="selectSuggestedTerm(activeIndex - 1)"
  />

  <button
    class="btn-search btn btn-primary"
    type="submit"
    name="search"
    [title]="'search.searchbox.button.title' | translate"
  >
    <!-- search button with icon -->
    <ng-container *ngIf="!configuration?.buttonText; else textBlock">
      <fa-icon [icon]="['fas', configuration?.icon || 'search']"></fa-icon>
    </ng-container>
    <!-- search button with text -->
    <ng-template #textBlock> {{ configuration.buttonText }} </ng-template>
  </button>

  <ul *ngIf="!isHidden && currentSearchTerm && results?.length" class="search-suggest-results">
    <li
      *ngFor="let result of results | slice: 0:configuration?.maxAutoSuggests; let liIndex = index"
      [class.active-suggestion]="isActiveSuggestedTerm(liIndex)"
      (mousedown)="submitSuggestedTerm(result.term)"
    >
      <button
        type="button"
        class="search-result"
        [innerHTML]="result.term | titlecase | ishHighlight: currentSearchTerm"
      ></button>
    </li>
  </ul>
</form>
