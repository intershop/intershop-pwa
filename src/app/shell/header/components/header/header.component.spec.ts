import { ComponentFixture, TestBed, async } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { NgbCollapseModule } from '@ng-bootstrap/ng-bootstrap';
import { StoreModule } from '@ngrx/store';
import { TranslateModule } from '@ngx-translate/core';
import { MockComponent } from 'ng-mocks';

import { FeatureToggleModule } from 'ish-core/feature-toggle.module';
import { IconModule } from 'ish-core/icon.module';
import { configurationReducer } from 'ish-core/store/configuration/configuration.reducer';
import { findAllIshElements } from 'ish-core/utils/dev/html-query-utils';
import { HeaderNavigationContainerComponent } from '../../containers/header-navigation/header-navigation.container';
import { LanguageSwitchContainerComponent } from '../../containers/language-switch/language-switch.container';
import { LoginStatusContainerComponent } from '../../containers/login-status/login-status.container';
import { MiniBasketContainerComponent } from '../../containers/mini-basket/mini-basket.container';
import { ProductCompareStatusContainerComponent } from '../../containers/product-compare-status/product-compare-status.container';
import { SearchBoxContainerComponent } from '../../containers/search-box/search-box.container';
import { UserInformationMobileComponent } from '../user-information-mobile/user-information-mobile.component';

import { HeaderComponent } from './header.component';

describe('Header Component', () => {
  let fixture: ComponentFixture<HeaderComponent>;
  let element: HTMLElement;
  let component: HeaderComponent;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [
        FeatureToggleModule,
        IconModule,
        NgbCollapseModule,
        RouterTestingModule,
        StoreModule.forRoot(
          { configuration: configurationReducer },
          { initialState: { configuration: { features: ['compare'] } } }
        ),
        TranslateModule.forRoot(),
      ],
      declarations: [
        HeaderComponent,
        MockComponent(HeaderNavigationContainerComponent),
        MockComponent(LanguageSwitchContainerComponent),
        MockComponent(LoginStatusContainerComponent),
        MockComponent(MiniBasketContainerComponent),
        MockComponent(ProductCompareStatusContainerComponent),
        MockComponent(SearchBoxContainerComponent),
        MockComponent(UserInformationMobileComponent),
      ],
    })
      .compileComponents()
      .then(() => {
        fixture = TestBed.createComponent(HeaderComponent);
        component = fixture.componentInstance;
        element = fixture.nativeElement;
      });
  }));

  it('should be created', () => {
    expect(component).toBeTruthy();
    expect(element).toBeTruthy();
    expect(() => fixture.detectChanges()).not.toThrow();
  });

  describe('rendering', () => {
    beforeEach(() => {
      fixture.detectChanges();
    });

    it('should render User Links on template', () => {
      expect(findAllIshElements(element)).toIncludeAllMembers([
        'ish-login-status-container',
        'ish-product-compare-status-container',
      ]);
    });
    it('should render Language Switch on template', () => {
      expect(findAllIshElements(element)).toContain('ish-language-switch-container');
    });

    it('should render Search Box on template', () => {
      expect(findAllIshElements(element)).toContain('ish-search-box-container');
    });

    it('should render Header Navigation on template', () => {
      expect(findAllIshElements(element)).toContain('ish-header-navigation-container');
    });

    it('should render normal header adequately for mobile devices', () => {
      component.deviceType = 'mobile';
      fixture.detectChanges();

      expect(element).toMatchSnapshot();
    });

    it('should render sticky header adequately for mobile devices', () => {
      component.deviceType = 'mobile';
      component.isSticky = true;
      fixture.detectChanges();

      expect(element).toMatchSnapshot();
    });

    it('should render normal header adequately for tablet devices', () => {
      component.deviceType = 'tablet';
      fixture.detectChanges();

      expect(element).toMatchSnapshot();
    });

    it('should render sticky header adequately for tablet devices', () => {
      component.deviceType = 'tablet';
      component.isSticky = true;
      fixture.detectChanges();

      expect(element).toMatchSnapshot();
    });

    it('should render normal header adequately for desktop', () => {
      component.deviceType = 'pc';
      fixture.detectChanges();

      expect(element).toMatchSnapshot();
    });

    it('should render sticky header adequately for desktop', () => {
      component.deviceType = 'pc';
      component.isSticky = true;
      fixture.detectChanges();

      expect(element).toMatchSnapshot();
    });
  });
});
