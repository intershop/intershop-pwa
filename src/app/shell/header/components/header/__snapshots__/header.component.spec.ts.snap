// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component rendering should render normal header adequately for desktop 1`] = `
<div class="header container">
  <ul class="user-links">
    <li>
      <ish-login-status-container
        data-testing-id="user-status-desktop"
      ></ish-login-status-container>
    </li>
    <li class="d-none d-md-block">
      <ish-product-compare-status-container
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status-container>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container
        data-testing-id="language-switch-desktop"
      ></ish-language-switch-container>
    </div>
    <ish-mini-basket-container
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket-container
    ><ish-mini-basket-container
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket-container>
  </div>
  <div class="search-container header-search-container">
    <ish-search-box-container data-testing-id="search-box-desktop"></ish-search-box-container>
  </div>
  <div class="logo-wrapper"></div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md collapse" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Component rendering should render normal header adequately for mobile devices 1`] = `
<div class="header container">
  <ul class="user-links">
    <li>
      <ish-login-status-container
        data-testing-id="user-status-desktop"
      ></ish-login-status-container>
    </li>
    <li class="d-none d-md-block">
      <ish-product-compare-status-container
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status-container>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container
        data-testing-id="language-switch-desktop"
      ></ish-language-switch-container>
    </div>
    <ish-mini-basket-container
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket-container
    ><ish-mini-basket-container
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket-container>
  </div>
  <div class="search-container header-search-container">
    <ish-search-box-container data-testing-id="search-box-desktop"></ish-search-box-container>
  </div>
  <div class="logo-wrapper"></div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md collapse" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Component rendering should render normal header adequately for tablet devices 1`] = `
<div class="header container">
  <ul class="user-links">
    <li>
      <ish-login-status-container
        data-testing-id="user-status-desktop"
      ></ish-login-status-container>
    </li>
    <li class="d-none d-md-block">
      <ish-product-compare-status-container
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status-container>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container
        data-testing-id="language-switch-desktop"
      ></ish-language-switch-container>
    </div>
    <ish-mini-basket-container
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket-container
    ><ish-mini-basket-container
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket-container>
  </div>
  <div class="search-container header-search-container">
    <ish-search-box-container data-testing-id="search-box-desktop"></ish-search-box-container>
  </div>
  <div class="logo-wrapper"></div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md collapse" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Component rendering should render sticky header adequately for desktop 1`] = `
<div class="header container">
  <ul class="user-links">
    <li>
      <ish-login-status-container
        data-testing-id="user-status-desktop"
      ></ish-login-status-container>
    </li>
    <li class="d-none d-md-block">
      <ish-product-compare-status-container
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status-container>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container
        data-testing-id="language-switch-desktop"
      ></ish-language-switch-container>
    </div>
    <ish-mini-basket-container
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket-container
    ><ish-mini-basket-container
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket-container>
  </div>
  <div class="search-container header-search-container">
    <ish-search-box-container data-testing-id="search-box-desktop"></ish-search-box-container>
  </div>
  <div class="logo-wrapper"></div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md collapse" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Component rendering should render sticky header adequately for mobile devices 1`] = `
<div class="header container">
  <ul class="user-links">
    <li>
      <ish-login-status-container
        data-testing-id="user-status-desktop"
      ></ish-login-status-container>
    </li>
    <li class="d-none d-md-block">
      <ish-product-compare-status-container
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status-container>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container
        data-testing-id="language-switch-desktop"
      ></ish-language-switch-container>
    </div>
    <ish-mini-basket-container
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket-container
    ><ish-mini-basket-container
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket-container>
  </div>
  <div class="search-container header-search-container">
    <ish-search-box-container data-testing-id="search-box-desktop"></ish-search-box-container>
  </div>
  <div class="logo-wrapper"></div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md collapse" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;

exports[`Header Component rendering should render sticky header adequately for tablet devices 1`] = `
<div class="header container">
  <ul class="user-links">
    <li>
      <ish-login-status-container
        data-testing-id="user-status-desktop"
      ></ish-login-status-container>
    </li>
    <li class="d-none d-md-block">
      <ish-product-compare-status-container
        data-testing-id="compare-status-desktop"
      ></ish-product-compare-status-container>
    </li>
  </ul>
  <div class="header-utility">
    <a class="search-toggler"></a>
    <div class="language-switch-container d-none d-md-block">
      <ish-language-switch-container
        data-testing-id="language-switch-desktop"
      ></ish-language-switch-container>
    </div>
    <ish-mini-basket-container
      class="d-none d-md-block"
      data-testing-id="mini-basket-desktop"
      view="full"
      ng-reflect-ng-class="d-md-block"
      ng-reflect-view="full"
    ></ish-mini-basket-container
    ><ish-mini-basket-container
      data-testing-id="mini-basket-mobile"
      view="small"
      class="d-md-none"
      ng-reflect-ng-class="d-md-none"
      ng-reflect-view="small"
    ></ish-mini-basket-container>
  </div>
  <div class="search-container header-search-container">
    <ish-search-box-container data-testing-id="search-box-desktop"></ish-search-box-container>
  </div>
  <div class="logo-wrapper"></div>
  <button class="navbar-toggler" type="button">
    <span class="sr-only">common.button.navbarCollapsed.text</span>
    <div class="hamburger-menu"><span></span><span></span><span></span></div>
  </button>
  <div class="global-nav-container">
    <nav class="global-nav navbar-expand-md collapse" ng-reflect-collapsed="true">
      <div class="main-navigation" role="navigation">
        <ish-header-navigation-container></ish-header-navigation-container
        ><ish-user-information-mobile></ish-user-information-mobile>
      </div>
    </nav>
  </div>
</div>
`;
